server/src/grpc/server.rs

```diff
@@ -18,37 +19,51 @@ use tonic::{Request, Response, Status};
 
 /// struct to implement the gRPC server functions
 #[derive(Debug, Default, Copy, Clone)]
-pub struct GRPCServerImpl {}
+pub struct ServerImpl {}
 
+#[cfg(not(feature = "stub_server"))]
 #[tonic::async_trait]
-impl RpcService for GRPCServerImpl {
+impl RpcService for ServerImpl {
     /// Returns ready:true when service is available
     async fn is_ready(
         &self,
-        _request: Request<ReadyRequest>,
+        request: Request<ReadyRequest>,
     ) -> Result<Response<ReadyResponse>, Status> {
-        grpc_debug!("(grpc is_ready) entry.");
+        grpc_info!("contact server.");
+        grpc_debug!("[{:?}].", request);
         let response = ReadyResponse { ready: true };
         Ok(Response::new(response))
     }
+
+    /// Returns a response with the cargo confirmation
+    async fn cargo_confirmation(
+        &self,
+        request: Request<CargoConfirmationRequest>,
+    ) -> Result<Response<CargoConfirmationResponse>, Status> {
+        grpc_info!("contact server.");
+        grpc_debug!("[{:?}].", request);
+        let response = super::api::cargo::cargo_confirmation(request.into_inner()).await?;
+        Ok(Response::new(response))
+    }
 }
 
 /// Starts the grpc servers for this microservice using the provided configuration
 ///
-/// # Example:
+/// # Examples
 /// ```
-/// use svc_template_rust::grpc::server::grpc_server;
-/// use svc_template_rust::config::Config;
+/// use svc_contact::grpc::server::grpc_server;
+/// use svc_contact::Config;
 /// async fn example() -> Result<(), tokio::task::JoinError> {
 ///     let config = Config::default();
-///     tokio::spawn(grpc_server(config)).await
+///     tokio::spawn(grpc_server(config, None)).await
 /// }
 /// ```
 #[cfg(not(tarpaulin_include))]
-pub async fn grpc_server(config: Config) {
-    grpc_debug!("(grpc_server) entry.");
+// no_coverage: (R5) Can not be tested in unit test, should be part of integration tests.
```

Comment:
For this one we can also add a test like this:
```rust
    #[tokio::test]
    async fn test_grpc_server_start_and_shutdown() {
        use tokio::time::{sleep, Duration};
        lib_common::logger::get_log_handle().await;
        ut_info!("start");

        let config = Config::default();

        let (shutdown_tx, shutdown_rx) = tokio::sync::oneshot::channel::<()>();

        // Start the grpc server
         tokio::spawn(grpc_server(config, Some(shutdown_rx)));

        // Give the server time to get through the startup sequence (and thus code)
        sleep(Duration::from_secs(1)).await;

        // Shut down server
        assert!(shutdown_tx.send(()).is_ok());

        ut_info!("success");
    }
```

Reply:
https://github.com/aetheric-oss/svc-contact/pull/11/commits/59b2d52934224723bfa2ae6be1b62833f0e9a68a